packages/nuxt/src/core/nuxt.ts

```diff
@@ -126,7 +126,10 @@ async function initNuxt (nuxt: Nuxt) {
 
   // Automatically register user modules
   for (const config of nuxt.options._layers.map(layer => layer.config)) {
-    const userModules = await resolveFiles(config.srcDir, `${config.dir?.modules || 'modules'}/*{${nuxt.options.extensions.join(',')}}`)
+    const userModules = await resolveFiles(config.srcDir, [
+      `${config.dir?.modules || 'modules'}/*{${nuxt.options.extensions.join(',')}}`,
+      `${config.dir?.modules || 'modules'}/*/{index,module}{${nuxt.options.extensions.join(',')}}`
```

Comment:
Thinking more, one other possible issue would be when there are some utils in modules with index, they might be wrongly scanned. 

@Atinux Are you good if we support `modules/*` + `modules/[name]/module.ts` for starting? This at least is compatible with current module structure recommendations.

Reply:
```suggestion
      `${config.dir?.modules || 'modules'}/*/module.{${nuxt.options.extensions.join(',')}}`
```

Reply:
I don't think it has the same constraints because you won't be importing from the module directly, I imagine?

Reply:
Indeed, only making it more future proof in case that we came with something (also if plan is to market it as build-your-module-then-extract, this has less change steps for converting it into a package)

Reply:
It is a hard decision, but I will avoid having both `module` + `index` and decide with one.

For local module, I like the idea of `index.ts` because it implies that you are loading the directory entry.

And anyway we have the module starter to better explain why we choose `module.ts`. I think we should add an `index.ts` there to explain that we export utils.

Pushing an update to current PR for docs, `modules/*/index.ts` makes the more sense when explaining